engines/wintermute/base/base_object.cpp

```diff
@@ -502,6 +504,43 @@ bool BaseObject::scCallMethod(ScScript *script, ScStack *stack, ScStack *thisSta
 	}
 
 #ifdef ENABLE_WME3D
+	//////////////////////////////////////////////////////////////////////////
+	// SetShadowImage
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "SetShadowImage") == 0) {
+		stack->correctParams(1);
+		ScValue *val = stack->pop();
+
+		if (_shadowImage) {
+			_gameRef->_surfaceStorage->removeSurface(_shadowImage);
+			_shadowImage = nullptr;
+		}
+
+		if (val->isString()) {
+			_shadowImage = _gameRef->_surfaceStorage->addSurface(val->getString());
+			stack->pushBool(_shadowImage != nullptr);
+		} else {
+			stack->pushBool(true);
+		}
+
+		return STATUS_OK;
+	}
+
+	//////////////////////////////////////////////////////////////////////////
+	// GetShadowImage
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "GetShadowImage") == 0) {
+		stack->correctParams(0);
+
+		if (_shadowImage) {
+			stack->pushString(_shadowImage->getFileName());
+		} else {
+			stack->pushBool(true);
```

Comment:
true? That's a weird value to get for a missing shadow-image.

Reply:
Right, I must have misread this one here, should be pushNULL().